### DISCUSSION TITLE
Access Remotely

### DISCUSSION THREAD

**Discussion by Bigmack3000** [advplyr/audiobookshelf#3300] [Q&A]
Hey, quick question.  How do I set up abs for remote access?  The server is installed through docker on a synology nas. I have port 13378 forwarding to the synology ip from my router, but I still can't get access to it outside of local.  Thanks!

**Comment by nichwall** [comment]
Before discussing your problem in more detail:

You will want to set up a reverse proxy (a simple one is NGINX Proxy Manager, it runs in Docker) instead of just port forwarding. Reverse Proxies will handle SSL termination, allowing you to use an encrypted connection instead of doing everything over plain text on the internet. Reverse Proxies are also reviewed by many more security minded people and handle common exploits to help protect your network versus what any individual application does (ABS or otherwise). A reverse proxy is not a silver bullet to handle everything, but is at least a recommended "bare minimum".

Another option is to set up a VPN to your home network using something like TailScale or WireGuard to allow for remote connections without opening ports to the internet, but this does require set up per client instead of just having a domain or IP you can type in to any computer (more secure than a reverse proxy).

Back to your original question:
Are you trying to connect to the server using the outside IP/port from inside your network or outside?

You may need to adjust your firewall settings to allow connections from outside your network.

**Reply by nichwall** [reply]
To clarify, with a reverse proxy you would still be forwarding a port from your router to the reverse proxy (which then routes to the ABS instance), but adds more security.

**Comment by Bigmack3000** [comment]
Thank you!  reverse proxy got it all working! I was able to set that up with a ddns on my synology, and now i can use the app anywhere i drive.

**Comment by jeeftor** [comment]
Tailscale is the easy mode option .... 

I use Tailscale + Cloudflare zero-trust tunnel

**Reply by strykedits** [reply]
I second Tailscale.    I also have a Synology NAS and Audiobookshelf, and as long as tailscale is downloaded and turned on, it works on all my devices.   I use it everyday with my iPhone via an app call Plappa.   It is an awesome app, like Bookplayer (my previous favorite book app), but works with my Synology.   So if you have an iPhone, download Tailscale and use Plappa.    There are a couple really good tailscale vids on YouTube, and it's shockingly easy to use.   No more crazy coding or reverse proxy needed, and it's free if you are a single user.

**Reply by jeeftor** [reply]
Plappa is great. CarPlay too
